### PR TITLE
Enable secrets in db resource group manager

### LINKED ISSUE #10286: Resource Group FlyWay Migration does not work with Trino Secrets (Server doesnt start)
Hello,

We currently use (secrets)[https://trino.io/docs/current/security/secrets.html] to manage our secrets within the Trino Environment. This has been working fine until our recent upgrade to 0.365. We noticed that there is an issue with the flyway migration. We know that a (change)[https://github.com/trinodb/trino/commit/cbd1027327401c708222e4378210f80d306155c3] was done for 0.364. We tried out the 0.365 version using secrets and found out the Trino Server is unable to start. The follow is the errors that we get.

```
021-12-13T18:40:43.617Z	INFO	main	io.trino.plugin.resourcegroups.db.FlywayMigration	Performing migrations...
2021-12-13T18:40:43.703Z	ERROR	main	io.trino.server.Server	Unable to obtain connection from database (jdbc:mysql://${ENV:CREDENTIALS_ENDPOINT}/${ENV:RESOURCE_GROUP_DB_NAME}?user=${ENV:CREDENTIALS_PRESTOPROXY_DB_USERNAME}&password=******************************************&useSSL=false) for user 'null': Cannot load connection class because of underlying exception: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 08001
Error Code : 0
Message    : Cannot load connection class because of underlying exception: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.

org.flywaydb.core.internal.exception.FlywaySqlException: Unable to obtain connection from database (jdbc:mysql://${ENV:CREDENTIALS_ENDPOINT}/${ENV:RESOURCE_GROUP_DB_NAME}?user=${ENV:CREDENTIALS_PRESTOPROXY_DB_USERNAME}&password=******************************************&useSSL=false) for user 'null': Cannot load connection class because of underlying exception: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 08001
Error Code : 0
Message    : Cannot load connection class because of underlying exception: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.

	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:67)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:70)
	at org.flywaydb.core.Flyway.execute(Flyway.java:504)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:170)
	at io.trino.plugin.resourcegroups.db.FlywayMigration.migrate(FlywayMigration.java:38)
	at io.trino.plugin.resourcegroups.db.DbResourceGroupConfigurationManagerFactory.create(DbResourceGroupConfigurationManagerFactory.java:42)
	at io.trino.execution.resourcegroups.InternalResourceGroupManager.setConfigurationManager(InternalResourceGroupManager.java:163)
	at io.trino.execution.resourcegroups.InternalResourceGroupManager.loadConfigurationManager(InternalResourceGroupManager.java:147)
	at io.trino.server.Server.doStart(Server.java:131)
	at io.trino.server.Server.lambda$start$0(Server.java:78)
	at io.trino.$gen.Trino_365_1____20211213_184022_1.run(Unknown Source)
	at io.trino.server.Server.start(Server.java:78)
	at io.trino.server.TrinoServer.main(TrinoServer.java:38)
Caused by: java.sql.SQLNonTransientConnectionException: Cannot load connection class because of underlying exception: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:110)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:73)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:79)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:131)
	at com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:221)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:268)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnection(DriverDataSource.java:232)
	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:52)
	... 12 more
Caused by: com.mysql.cj.exceptions.UnableToConnectException: Cannot load connection class because of underlying exception: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105)
	... 16 more
Caused by: com.mysql.cj.exceptions.WrongArgumentException: Failed to parse the host:port pair '${ENV:CREDENTIALS_ENDPOINT}'.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105)
	at com.mysql.cj.conf.ConnectionUrlParser.parseHostPortPair(ConnectionUrlParser.java:502)
	at com.mysql.cj.conf.ConnectionUrlParser.buildHostInfoResortingToGenericSyntaxParser(ConnectionUrlParser.java:440)
	at com.mysql.cj.conf.ConnectionUrlParser.parseAuthoritySegment(ConnectionUrlParser.java:269)
	at com.mysql.cj.conf.ConnectionUrlParser.parseAuthoritySection(ConnectionUrlParser.java:183)
	at com.mysql.cj.conf.ConnectionUrlParser.getHosts(ConnectionUrlParser.java:653)
	at com.mysql.cj.conf.ConnectionUrl$Type.getConnectionUrlInstance(ConnectionUrl.java:194)
	at com.mysql.cj.conf.ConnectionUrl.getConnectionUrlInstance(ConnectionUrl.java:280)
	at com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:195)
	... 15 more
Caused by: java.lang.NumberFormatException: For input string: "CREDENTIALS_ENDPOINT}"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at com.mysql.cj.conf.ConnectionUrlParser.parseHostPortPair(ConnectionUrlParser.java:500)
	... 22 more
```

### PR DESCRIPTION
**Pull request by posulliv**
## General information

Is this change a fix, improvement, new feature, refactoring, or other?

A fix to the database resource group manager.

Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Update to the database resource group manager to enable the use of secrets in the configuration file.

How would you describe this change to a non-technical end user or system administrator?

## Related issues, pull requests, and links
Fixes #10286

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```
# Section
* Fix some things. ({issue}`5678`)
```

### CODE REVIEW

**Comment by posulliv** [comment]
CC @akashkatipally

**Comment by akashkatipally** [comment]
lgtm

File: plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/db/DbResourceGroupConfigurationManagerFactory.java [hunk 1]
```diff
@@ -39,6 +40,7 @@ public String getName()
     @Override
     public ResourceGroupConfigurationManager<?> create(Map<String, String> config, ResourceGroupConfigurationManagerContext context)
     {
+        config = ConfigurationUtils.replaceEnvironmentVariables(config);
```
Comment (findepi) [inline plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/db/DbResourceGroupConfigurationManagerFactory.java:43]: import statically

**Review by findepi — APPROVED** [review]

**Review by hashhar — APPROVED** [review]
Can we test this?

**Comment by hashhar** [comment]
```
Error:  io.trino.memory.TestMemoryManager.testOutOfMemoryKiller  Time elapsed: 6.675 s  <<< FAILURE!
java.lang.AssertionError: expected object to not be null
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.assertNotNull(Assert.java:423)
	at org.testng.Assert.assertNotNull(Assert.java:408)
	at io.trino.memory.TestMemoryManager.waitForQueryToBeKilled(TestMemoryManager.java:156)
	at io.trino.memory.TestMemoryManager.testOutOfMemoryKiller(TestMemoryManager.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Unrelated. Wasn't able to reproduce with 100 runs.

**Comment by findepi** [comment]
@hashhar issue?

**Comment by findepi** [comment]
> Can we test this?


You can't really set env vars in a test, so that would be.... a product test?
Not worth it.

**Comment by electrum** [comment]
I don’t understand why this is needed. The code already uses `Bootstrap` which replaces secrets.

**Comment by findepi** [comment]
> I don’t understand why this is needed. The code already uses `Bootstrap` which replaces secrets.

@electrum see https://github.com/trinodb/trino/pull/12975

**Pull request merged**